test/signalrclient-e2e-tests/connection_tests.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+#include "url.h"
+
+TEST(connection_tests, connection_status_start_stop)
+{
+    auto conn = std::make_shared<signalr::connection>(url + U("raw-connection"));
+    auto test_event = std::make_shared<pplx::event>();
```

Comment:
I know that gtest does not support timeout. I am saying that this event here is useless because you are running the code synchronously. So `ASSERT_FALSE(test_event->wait(15000));` can only fail if you have an exception and you won't signal the event. However if you have an exception then the test will be terminated because of `ASSERT_TRUE(false);` in the handler. As I pointed below - even if you didn't have the `try...catch` (which I think you should actually get rid of) `gtest` will catch the exception and handle it and you would never get to the last line. The event is just not needed here.